src/components/SectionCheckbox.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { useCallback } from "react";
+import StudentCheckbox from "./StudentCheckbox";
+
+const SectionCheck = ({
+	sectionIndex,
+	sectionName,
+	students,
+	sectionChecked,
+	handleSectionCheck,
+	handleStudentCheck,
+}) => {
+
+	const handleCheck = useCallback((studentIndex) => {
+		handleStudentCheck(studentIndex, sectionIndex);
+	}, []);
+
+	const handleSecCheck = useCallback((sectionIndex) => {
+		handleSectionCheck(sectionIndex);
+	}, []);
+
+	return (
+		<div className="section-container">
+			<div className="section-tab">
+				<input
+					type="checkbox"
+					name="section"
+					checked={sectionChecked}
+					onChange={() => handleSecCheck(sectionIndex)}
+				/>
+				<label>{sectionName}</label>
+			</div>
+			<div className="student-container">
+				{students.map((student, index) => (
```

Comment:
Add data check. 

`{students?.map((student, index) => (`

Reply:
Sure will make the changes.